firebase-segmentation/src/main/java/com/google/firebase/segmentation/FirebaseSegmentation.java

```diff
@@ -55,7 +79,161 @@ public static FirebaseSegmentation getInstance(@NonNull FirebaseApp app) {
     return app.get(FirebaseSegmentation.class);
   }
 
-  Task<Void> setCustomInstallationId(String customInstallationId) {
-    return Tasks.forResult(null);
+  Task<Void> setCustomInstallationId(@Nullable String customInstallationId) {
```

Comment:
should the return type be `@NonNull public Task<Void>` instead?

Reply:
Also what are the thread-safety/concurrency requirements for this method(including the ones it calls)? I looks like multiple calls to this from different threads could lead to races

Reply:
1)  Yes, definitely it should be nonnull public

2)  Change it to be synchronized